toolkit/tools/internal/safechroot/safechroot.go

```diff
@@ -229,7 +243,21 @@ func (c *Chroot) Initialize(tarPath string, extraDirectories []string, extraMoun
 	// mount is only supported in regular pipeline
 	if buildpipeline.IsRegularBuild() {
 		// Create kernel mountpoints
-		allMountPoints := append(defaultMountPoints(), extraMountPoints...)
+		allMountPoints := []*MountPoint{}
+
+		for _, mountPoint := range extraMountPoints {
+			if mountPoint.mountBeforeDefaults {
+				allMountPoints = append(allMountPoints, mountPoint)
+			}
+		}
+
+		allMountPoints = append(allMountPoints, defaultMountPoints()...)
```

Comment:
When building an image from scratch, the rootfs is just a normal directory. But when modifying and existing image, the rootfs has to be mounted before all the other partitions. Hence this change.